set.mm

```diff
@@ -412491,6 +412575,135 @@ about equation (2) in section I.1 in [Bollobas] p. 4.  (Contributed by
 #*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
   Walks, paths and cycles
 #*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
+
+  A "walk" in a graph is usually defined for simple graphs, multigraphs or even
+  pseudographs as "alternating sequence of vertices and edges x0 , e1 , x1 ,
+  e2 , ... , e(l) , x(l) where e(i) = x(i-1)x(i), 0<i<=l.", see definition of
+  [Bollobas] p. 4, or "A walk (of length k) in a graph is a non-empty
+  alternating sequence v_0 e_0 v_1 e_1 ... e_k-1 v_k of vertices and edges in G
+  such that e_i = { v_i , v_i+1 } for all i < k.", see definition of [Diestel]
+  p. 10.
+
```

Comment:
So why not define a walk in G as a function
f : [[0, 2n]] --> ( Edge(G) u. Vertex(G) ) such that for all 0<=k<=n-1, f(2k) e. Vertex(G) and f(2k+1) e. Edge(G, f(2k), f(2k+2))
(where the last set is the set of edges in G between to given vertices; I do not know how it is named in set.mm) ?

Reply:
There is no definition for the set of edges between (two) vertices in set.mm, as far as I know. Therefore, f(2k+1) e. Edge(G, f(2k), f(2k+2)) would have to be replaced by ( f(2k+1) e. Edge(G) /\ { f(2k), f(2k+2) } C_ f(2k+1) ). But is it really useful to mix both sequences into one function? I could imagine that working with such a definition, although it might be more elegant, would be very tedious.

Since a lot of theorems (especially the Koenigsbrg bridge theorem and the friendship theorem) are based on the current definition, which was already used/introduced by @digama0, it would be a huge undertaking. Maybe a study of this approach could be started in a mathbox.

Reply:
Ok. Maybe add to the head comment something like:
Therefore, to stay close to the textbook definition, one could define a walk as a function f : [0...2n] --> ( Edge(G) u. Vertex(G) ) such that for all 0<=k<=n, f(2k) e. Vertex(G) and for all 0<=k<=n-1, ( f(2k+1) e. Edge(G) /\ { f(2k), f(2k+2) } C_ f(2k+1) ) (in "pseudocode").  However, this definition would be uneasy to use, therefore, we decide to define walks as pairs of functions f from { 1 , ... , n } and p from { 0 , ... , n }, where f enumerates...

It might indeed be interesting to state the alternate definition in a mathbox and prove equivalence:
```
function f into (Vertex u. Edges)      <->  interlaced functions v and e
f                                      |->  (f o. (x |-> 2x) , f o. (x |-> 2x+1)) 
(x |-> if(x even, v(x/2), e((x-1)/2))) <-|  (v, e)
```

Reply:
I will add a hint to this alternate definition in my next PR.

Reply:
> Ok. Maybe add to the head comment something like: Therefore, to stay close to the textbook definition, one could define a walk as a function f : [0...2n] --> ( Edge(G) u. Vertex(G) ) such that for all 0<=k<=n, f(2k) e. Vertex(G) and for all 0<=k<=n-1, ( f(2k+1) e. Edge(G) /\ { f(2k), f(2k+2) } C_ f(2k+1) ) (in "pseudocode").

I would challenge the claim that this is in fact closer to the textbook definition if such a path is notated as `v_0 e_0 v_1 e_1 ...`. In a textbook presentation I can't imagine a single sequence where the type of the object changes on every alternate index being less confusing in any way.

Reply:
@digama0 : you're probably right, but... pistol or sword ? Where shall we meet at dawn ? I choose @avekens as my second.

Reply:
If we're dueling, then as my evidence I would point to [wikipedia]{https://en.wikipedia.org/wiki/Path_(graph_theory)) and [MathWorld](https://mathworld.wolfram.com/Walk.html), which both use two sequences `v_i` and `e_i` (although the wikipedia version differs slightly in taking the v_i sequence as existential). Neither one even mentions the idea of an even/odd sequence such as you have proposed. Do you have a source?

Reply:
As I wrote above, you're probably right, but dueling is no fun without a dose of bad faith ! (and irony is harder to convey in words)